drivers/include/periph/gpio_ll.h

```diff
@@ -95,36 +95,55 @@ typedef uintptr_t gpio_port_t;
 
 #ifdef DOXYGEN
 /**
- * @brief   Get the @ref gpio_port_t value of the port identified by @p num
+ * @brief   Get the @ref gpio_port_t value of the port labeled 0.
  *
- * @note    If @p num is a compile time constant, this is guaranteed to be
- *          suitable for a constant initializer.
+ * @note    For MCUs that use letters instead of numbers, this will be an alias
+ *          for @ref GPIO_PORT_A
+ * @note    Some MCUs will not start with Port 0 / Port A, but rather with
+ *          Port 1 (e.g. MSP430) or Port B (e.g. ATmega328P). It will be
+ *          undefined when unavailable
+ * @note    There will also be `GPIO_PORT_1`, `GPIO_PORT_2`, etc. when there
+ *          are corresponding GPIO ports in hardware.
+ */
+#define GPIO_PORT_0  implementation_specific
+/**
+ * @brief   Same as @ref GPIO_PORT_0, but only provided on MCUs that do use
+ *          letters to refer to GPIO ports.
  *
- * Typically this will be something like `(GPIO_BASE_ADDR + num * sizeof(struct
- * vendor_gpio_reg))`
+ * @note    The macro with the numeric naming scheme will always be defined,
+ *          the alphabetic naming scheme only if the MCU does uses this.
  */
-#define GPIO_PORT(num)  implementation_specific
-#endif
+#define GPIO_PORT_A  implementation_specific
```

Comment:
Why not always define

```suggestion
#ifdef GPIO_PORT_0
#define GPIO_PORT_A  GPIO_PORT_0
#endif
```

Reply:
My plan was originally to just provide the alphabetic enumeration only on MCUs that do so, so that this could also be used to track what the user friendly way of printing GPIOs would be (e.g. PA1 or P0.1).

But I agree that this was not the smartest idea I had. An `#define GPIO_PORTS_USE_ALPHABETICAL_ENUMERATION 1` would be more obvious.

Reply:
I updated it as suggested.